src/intelligence_layer/connectors/data/models.py

```diff
@@ -0,0 +1,89 @@
+import io
+from datetime import datetime
+from typing import Any, Optional
+
+from pydantic import BaseModel, ConfigDict
+from pydantic.alias_generators import to_camel, to_snake
+
+
+class BaseDataModel(BaseModel):
+    model_config = ConfigDict(alias_generator=to_camel)
+
+
+class DataRepository(BaseDataModel):
+    """Data Repository model.
+
+    Attributes:
+    repository_id: Repository ID that identifies the repository(group of datasets)
+    name: Name of the repository
+    mutable: Indicates if the datasets in the repository are mutable or not
+    mediaType: Media type of the data: application/json, application/csv, etc.
+    modality: Modality of the data: image, text, etc.
+    createdAt: Datetime when the repository was created
+    updatedAt: Datetime when the repository was updated
+    """
+
+    repository_id: str
+    name: str
+    mutable: bool
+    media_type: str
+    modality: str
+    created_at: datetime
+    updated_at: datetime
+
+
+class DataRepositoryCreate(BaseDataModel):
```

Comment:
You should add in all of the models the recommended Pydantic 2 way of using configs and alias:

`model_config = ConfigDict(alias_generator=AliasGenerator(serialization_alias=to_camel), arbitrary_types_allowed=True)`

The `ConfigDict` and `AliasGenerator` imports are from Pydantic.